### ISSUE TITLE
LVM segfault

### ISSUE THREAD

**Issue by mmcgr** [souffle-lang/souffle#964]
64bit souffle segfaults in some, rare, cases, e.g., http://plang1.it.usyd.edu.au/jenkins/job/Master-Ubuntu-LTS-64bit-domains/266/console

Backtrace
```
#0  0x00005617e04da369 in __gnu_cxx::new_allocator<long>::construct<long, long const&> (this=0x7ffce29ef640, __p=0x5617e253a990, __args#0=<error reading variable>) at /usr/include/c++/7/ext/new_allocator.h:136
#1  0x00005617e04c3674 in std::allocator_traits<std::allocator<long> >::construct<long, long const&> (__a=..., __p=0x5617e253a990, __args#0=<error reading variable>) at /usr/include/c++/7/bits/alloc_traits.h:475
#2  0x00005617e052fd58 in std::deque<long, std::allocator<long> >::push_back (this=0x7ffce29ef640, __x=<error reading variable>) at /usr/include/c++/7/bits/stl_deque.h:1547
#3  0x00005617e052bec5 in std::stack<long, std::deque<long, std::allocator<long> > >::push (this=0x7ffce29ef640, __x=<error reading variable>) at /usr/include/c++/7/bits/stl_stack.h:226
#4  0x00005617e06722e8 in souffle::Interpreter::execute (this=0x5617e253a100, codeStream=std::unique_ptr<souffle::LVMCode> = {...}, ctxt=..., ip=130) at Interpreter.cpp:131
#5  0x00005617e0671a49 in souffle::Interpreter::executeMain (this=0x5617e253a100) at Interpreter.cpp:74
#6  0x00005617e046bcc6 in souffle::main (argc=6, argv=0x7ffce29f0e08) at main.cpp:530
#7  0x00005617e046ed9d in main (argc=6, argv=0x7ffce29f0e08) at main.cpp:635
```
frame 4 is
```cpp
130                 case LVM_ElementAccess:
131                     stack.push(ctxt[code[ip + 1]][code[ip + 2]]);
132                     ip += 3;
133                     break;
```
ctxt is
```cpp
(souffle::InterpreterContext &) @0x7ffce29ef800: {_vptr.InterpreterContext = 0x5617e0b98210 <vtable for souffle::InterpreterContext+16>, data = std::vector of length 2, capacity 2 = {0x5617e2552250, 0x0}, returnValues = 0x0,
  returnErrors = 0x0, args = 0x0}
```

i.e. ctxt[1] is a nullptr and we try to access ctxt[1][1]

**Comment by b-scholz** [comment]
Sounds like an integer overflow due to 32/64 bit mix-up.

**Comment by mmcgr** [comment]
I have duplicated it with g++-6 on 18.04. The issue seems to have been fixed by #991 .
